### DISCUSSION TITLE
Migrating from Cloudlog to Wavelog

### DISCUSSION THREAD

**Discussion by DJLax5** [wavelog/wavelog#1315] [Q&A] [answered]
Hi everyone!

I've been using Cloudlog for the past few years. But a few features from wavelog seem promising. So, I was thinking about making the transition. 

However I have well over 30 Station Locations, with thousands of QSOs and hundreds of Scanned QSL cards in my instance. It would be a huge pain to migrate all via ADIF due to the many Station locations and especially the linkage of my QSL card images would be lost (or is there a global import way??).

I've read that migrating through changing the version number in the migration.php and changing the git remote should work. Does that also work for the QSL card images? 

I've seen that you guys also have your own docker image. Using that would be amazing. 
I could connect your docker image to my existing Cloudlog database, but I suspect bad things would happen :D

Is there a smart way to do this? 

Thanks in advance,
Fabian

**Comment by HB9HIL** [comment, marked as answer]
Hello Fabian

Thanks for asking for support. You're right. There is a Migration Wiki which describes how to migrate from Cloudlog to Wavelog. This is for `git` install to `git` install or for file-based to file-based. 

There is currently no official way/tutorial to migrate from a `git` based Cloudlog Installation to a Dockerized Wavelog Installation. Sure it's doable but requires some knowledge about databases and the way how Docker works. 

The simplest Way would be to migrate to a `git` Installation as desribed in the Wiki. You won't loose your QSL Card images through that. But a few months ago we changed the way how QSL/eQSL Cards are stored within Wavelog. Here is a solution implemented in the Debug View which migrates the data for you. 

Four things are important:
- MAKE A BACKUP BEFORE YOU DO ANYTHING 😄 Database and Filesystem
- Migrate with the Wiki [here](https://github.com/wavelog/Wavelog/wiki/How-to-migrate-Cloudlog-to-Wavelog). Follow **each** step and **read each** step. This is important.
- After migration compare the `config.sample.php` with your `config.php`
- To migrate the QSL card follow [this wiki](https://github.com/wavelog/wavelog/wiki/Centralized-Userdata). You only have to follow steps 6 to 9.
- **Never** edit wavelog.php

If you are not sure with something or you are facing issues, contact me directly via hb9hil@wavelog.org. I did not test the QSL migrating for quite a while so would be good to have a feedback here. 

Vy 73 de HB9HIL
Also Fabian 😉

**Reply by DJLax5** [reply]
Thanks for the quick reply! I will test it on the weekend,  then I have time :D

**Reply by DJLax5** [reply]
Yup, that pretty much worked right like you told me. I had some server issues with php, but those were my fault.. But yes, now all my station locations, QSOs and QSLs are still where they are supposed to be!

**Reply by HB9HIL** [reply]
Awesome! Welcome to Wavelog :-) if there are any issue don't hestitate to ask the community in the discussions or create an issue if you find any bugs.
